File: split/getrtoken.pas
procedure getrtoken;
label
  20;
begin
  20:
    repeat
      gettoken;
    until curtok <> 2592;
  if (curcs = 0) or (curcs > 2614) then
  begin
    begin
      if interaction = 3 then;
      printnl(262);
      print(1183);
    end;
    begin
      helpptr := 5;
      helpline[4] := 1184;
      helpline[3] := 1185;
      helpline[2] := 1186;
      helpline[1] := 1187;
      helpline[0] := 1188;
    end;
    if curcs = 0 then
      backinput;
    curtok := 6709;
    inserror;
    goto 20;
  end;
end;

File: cpp/getrtoken.cpp
#include "getrtoken.h"
#include "gettoken.h"
#include "printnl.h"
#include "print.h"
#include "backinput.h"
#include "inserror.h"
#include "texte.h"

void getrtoken(void)
{
	while (true)
	{
		do
			gettoken();
		while (curtok == spacer*0x01'00+' ');
		if (curcs == 0 || curcs > frozen_control_sequence)
		{
			printnl("! ");
			print("Missing control sequence inserted");
			helpptr = 5;
			helpline[4] = "Please don't say `\\def cs{...}', say `\\def\\cs{...}'.";
			helpline[3] = "I've inserted an inaccessible control sequence so that your";
			helpline[2] = "definition will be completed without mixing me up too badly.";
			helpline[1] = "You can recover graciously from this error, if you're";
			helpline[0] = "careful; see exercise 27.2 in The TeXbook.";
			if (curcs == 0)
				backinput();
			curtok = frozen_protection+cs_token_flag;
			inserror();
			continue;
		}
		break;
	}
}
